### PR TITLE
Add support for Terraform 0.12

### LINKED ISSUE #7: Migrate to Terraform 0.12
As other modules under `github.com/terraform-google-modules` are moving to TF v0.12 in recent weeks, this one should too.

I'm happy to help out with this one if required.

### PR DESCRIPTION
**Pull request by Dev25**
Fixes #7

### CODE REVIEW

**Comment by Dev25** [comment]
Couple things to note primarily with the `make` scripts/tests, most of them are currently broken atm
- `make check` fails since `terraform validate -check-variables` flag  is no more in 0.12
- `make generate_docs` is broken since terraform-docs doesn't support 0.12 yet, other modules have moved to using a helper script with 0.12 support terraform_docs.sh

According to https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/209 there is some work ongoing to help fix this across all modules. The question is should i try to fix these issues (mainly from copying existing 0.12 modules) or just wait for the docker move mentioned in the issue?

File: main.tf [hunk 1]
```diff
@@ -14,65 +14,77 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.11.0"
-}
-
 locals {
   prefix = "${var.prefix == "" ? "" : "${var.prefix}-"}${lower(var.location)}"
```
[the remark this answers is not in the document]
Comment (Dev25) [reply main.tf]: Done. This wasn't used before so i've also fixed that in this PR. Old behaviour meant that with a empty `prefix` your final bucket name would be `-{name}`.

New behaviour:
```
# prefix=""
+ name               = "first"

# prefix="foo"
+ name               = "foo-eu-first"
```

**Comment by Dev25** [comment]
@aaron-lane I've updated this PR to also migrate to the new docker image stuff for developer testing as per https://github.com/terraform-google-modules/terraform-google-network/pull/58

This repo doesn't have CI setup but i can confirm `make docker_test_integration` and `make docker_test_integration` pass

**Comment by aaron-lane** [comment]
@Dev25 I apologize for the delay! This work was unfortunately duplicated by #9, which has already been merged and released, so I'm going to close this. We do appreciate your interest in contributing, though!

**Pull request closed without merging**